src/coreclr/src/pal/src/thread/process.cpp

```diff
@@ -3405,6 +3422,30 @@ FlushProcessWriteBuffers()
     }
     else
     {
+#if defined(TARGET_OSX)
+        mach_msg_type_number_t cThreads;
```

Comment:
For x64, this should be just a fallback for case the initialization of `s_helperPage` (`g_helperPage` in the standalone GC case) failed due to `mlock` failure. The page protection change way is still going to be more performant. The InitializeFlushProcessWriteBuffers would need to be slightly modified to e.g. clear the `s_helperPage` in case it fails to `mlock` so that we can detect that case here.